packages/dataviews/src/view-table.js

```diff
@@ -306,6 +307,80 @@ function WithSeparators( { children } ) {
 		) );
 }
 
+function BulkSelectionCheckbox( { selection, onSelectionChange, data } ) {
+	const areAllSelected = selection && selection.length === data.length;
```

Comment:
Nit: We always initiate selection with an empty array, so no need for the first check, right?